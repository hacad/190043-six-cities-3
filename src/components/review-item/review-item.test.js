import React from "react";
import renderer from "react-test-renderer";
import ReviewItem from "./review-item.jsx";

it(`ReviewItem correctly renders after relaunch`, () => {
  const review = {
    "id": 1,
    "user": {
      "id": 17,
      "isPro": false,
      "name": `Emely`,
      "avatarUrl": `https://htmlacademy-react-2.appspot.com/six-cities/static/avatar/8.jpg`
    },
    "rating": 3,
    "comment": `The room was spacious and clean. The pool looked nothing like the photos and desparately needs a clean. The sauna and spa were closed for lunar new year holiday.`,
    "date": new Date(`2019-12-03T14:11:47.471Z`)
  };
  const reviewItem = renderer.create(
      <ReviewItem review={review}/>
  ).toJSON();

  expect(reviewItem).toMatchSnapshot();
});
