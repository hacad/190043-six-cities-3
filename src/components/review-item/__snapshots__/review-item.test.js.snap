// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewItem correctly renders after relaunch 1`] = `
<li
  className="reviews__item"
>
  <div
    className="reviews__user user"
  >
    <div
      className="reviews__avatar-wrapper user__avatar-wrapper"
    >
      <img
        alt="Reviews avatar"
        className="reviews__avatar user__avatar"
        height="54"
        src="https://htmlacademy-react-2.appspot.com/six-cities/static/avatar/8.jpg"
        width="54"
      />
    </div>
    <span
      className="reviews__user-name"
    >
      Emely
    </span>
  </div>
  <div
    className="reviews__info"
  >
    <div
      className="reviews__rating rating"
    >
      <div
        className="reviews__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "3%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <p
      className="reviews__text"
    >
      The room was spacious and clean. The pool looked nothing like the photos and desparately needs a clean. The sauna and spa were closed for lunar new year holiday.
    </p>
    <time
      className="reviews__time"
      dateTime={2019-12-03T14:11:47.471Z}
    />
  </div>
</li>
`;
